File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/commerceaccountgroupcommerceaccountrel/CommerceaccountgroupcommerceaccountrelService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.commerceaccountgroupcommerceaccountrel;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CommerceaccountgroupcommerceaccountrelService extends BaseService {

	public CommerceaccountgroupcommerceaccountrelService(Session session) {
		super(session);
	}

	public JSONObject addCommerceAccountGroupCommerceAccountRel(long commerceAccountGroupId, long commerceAccountId, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountGroupId", commerceAccountGroupId);
			_params.put("commerceAccountId", commerceAccountId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commerceaccountgroupcommerceaccountrel/add-commerce-account-group-commerce-account-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void deleteCommerceAccountGroupCommerceAccountRel(long commerceAccountGroupCommerceAccountRelId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountGroupCommerceAccountRelId", commerceAccountGroupCommerceAccountRelId);

			_command.put("/commerce.commerceaccountgroupcommerceaccountrel/delete-commerce-account-group-commerce-account-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject getCommerceAccountGroupCommerceAccountRel(long commerceAccountGroupId, long commerceAccountId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountGroupId", commerceAccountGroupId);
			_params.put("commerceAccountId", commerceAccountId);

			_command.put("/commerce.commerceaccountgroupcommerceaccountrel/get-commerce-account-group-commerce-account-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getCommerceAccountGroupCommerceAccountRels(long commerceAccountGroupId, int start, int end) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountGroupId", commerceAccountGroupId);
			_params.put("end", end);
			_params.put("start", start);

			_command.put("/commerce.commerceaccountgroupcommerceaccountrel/get-commerce-account-group-commerce-account-rels", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getCommerceAccountGroupCommerceAccountRelsCount(long commerceAccountGroupId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountGroupId", commerceAccountGroupId);

			_command.put("/commerce.commerceaccountgroupcommerceaccountrel/get-commerce-account-group-commerce-account-rels-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

}

File: windows/Liferay.SDK/Service/V73/Commerceaccountgroupcommerceaccountrel/CommerceaccountgroupcommerceaccountrelService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Commerceaccountgroupcommerceaccountrel
{
	public class CommerceaccountgroupcommerceaccountrelService : ServiceBase
	{
		public CommerceaccountgroupcommerceaccountrelService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> AddCommerceAccountGroupCommerceAccountRelAsync(long commerceAccountGroupId, long commerceAccountId, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountGroupId", commerceAccountGroupId);
			_parameters.Add("commerceAccountId", commerceAccountId);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccountgroupcommerceaccountrel/add-commerce-account-group-commerce-account-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetCommerceAccountGroupCommerceAccountRelAsync(long commerceAccountGroupId, long commerceAccountId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountGroupId", commerceAccountGroupId);
			_parameters.Add("commerceAccountId", commerceAccountId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccountgroupcommerceaccountrel/get-commerce-account-group-commerce-account-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceAccountGroupCommerceAccountRelsAsync(long commerceAccountGroupId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountGroupId", commerceAccountGroupId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccountgroupcommerceaccountrel/get-commerce-account-group-commerce-account-rels", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetCommerceAccountGroupCommerceAccountRelsCountAsync(long commerceAccountGroupId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountGroupId", commerceAccountGroupId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccountgroupcommerceaccountrel/get-commerce-account-group-commerce-account-rels-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task DeleteCommerceAccountGroupCommerceAccountRelAsync(long commerceAccountGroupCommerceAccountRelId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceAccountGroupCommerceAccountRelId", commerceAccountGroupCommerceAccountRelId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceaccountgroupcommerceaccountrel/delete-commerce-account-group-commerce-account-rel", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}
	}
}